Tests/CookiesTests.cs

```diff
@@ -0,0 +1,191 @@
+
+        cookiePage.Response.Headers.TryGetValue("Set-Cookie", out var cookieVariables);
+        cookieVariables.ToString().Should().Contain(".FindATuitionPartner.Consent=False");
+    }
+
+    [Fact]
+    public async void Cookies_Associated_With_FindATuitionPartner_Deleted_When_Set_To_False()
+    {
+        var cookiePage = await Fixture.GetPage<Cookies>()
+              .Execute(page =>
+              {
+                  page.PageContext = GetPageContext();
+                  page.Request.Cookies = MockRequestCookieCollection(".FindATuitionPartner", "true");
+                  page.Request.Cookies.ContainsKey("_ga").Should().BeTrue();
+                  page.Request.Cookies.ContainsKey("_gid").Should().BeTrue();
+                  var cookieOptions = new CookieOptions { Expires = DateTime.Today.AddMonths(12), HttpOnly = true, Secure = true, SameSite = Microsoft.AspNetCore.Http.SameSiteMode.Strict };
+                  page.Response.Cookies.Append("_ga", "hasValueToDelete", cookieOptions);
+                  page.Response.Cookies.Append("_gid", "hasValueToDelete", cookieOptions);
+                  page.Response.Headers.TryGetValue("Set-Cookie", out var cookieVariables);
+                  cookieVariables.ToString().Should().Contain("_ga=hasValueToDelete;");
+                  cookieVariables.ToString().Should().Contain("_gid=hasValueToDelete;");
+                  page.OnGet(false, true);
+                  return Task.FromResult(page);
+              });
+        cookiePage.Response.Headers.TryGetValue("Set-Cookie", out var cookieVariables);
+        cookieVariables.ToString().Should().Contain("_ga=;");
+        cookieVariables.ToString().Should().Contain("_gid=;");
+    }
+
+    [Fact]
+    public async void Cookies_OnPost_Model_Valid()
+    {
+        var cookiePage = await Fixture.GetPage<Cookies>()
+            .Execute(page =>
+            {
+                page.PageContext = GetPageContext();
+                page.OnPost();
+                return Task.FromResult(page);
+            });
+        cookiePage.Should().BeOfType<Cookies>();
+        cookiePage.ModelState.IsValid.Should().BeTrue();
+
+        var result = cookiePage.OnPost();
+        result.Should().BeOfType<RedirectToPageResult>();
+    }
+
+    [Fact]
+    public async void Cookies_OnPost_Model_Invalid()
+    {
+        var cookiePage = await Fixture.GetPage<Cookies>()
+           .Execute(page =>
+           {
+               page.PageContext = GetPageContext();
+               page.ModelState.AddModelError("cookies", "You must select an option");
+               page.OnPost();
+               return Task.FromResult(page);
+           });
+        cookiePage.Should().BeOfType<Cookies>();
+        cookiePage.ModelState.IsValid.Should().BeFalse();
+
+        cookiePage.ModelState.AddModelError("cookies", "You must select an option");
+        var result = cookiePage.OnPost();
+        result.Should().BeOfType<PageResult>();
+    }
+
```

Comment:
I don't see direct tests for OnGet preferencesSet or ReturnUrl. There's logic there for when to set it and it should be possible to check the value of ReturnUrl on the page following the call.

Reply:
Added tests for OnGet preferencesSet or ReturnUrl.